ddp2 and demon Front fix USER1 rom loading

arcadez2003 opened the pull request.

amadvance commented:
@arcadez2003 Now dmnfrnt behaves like kov2p. After showing the date it doesn't proceed further. Instead, ddp2 shows the following init error after the date.

![ddp2](https://github.com/user-attachments/assets/19fed9cc-f2af-4810-b8f5-fb59934eef60)

arcadez2003 commented:
i'll focus on DDP 2 as that one just hangs off our KOV 2 code which is confirmed to work.

Some things come to mind...

1) Some games PGM games dont like the IRQ4 to be active i'll need to check if ddp2 is one of them
2) Maybe we need those IRQ changes from when demon front was added
3) There is a change to the ARM7 when the above games were added maybe we need it here.
4) The ddp2 prot sim has a busy counter bypass hack maybe we can use it for a simple fix.

arcadez2003 commented:
@amadvance without rabbiting on check this commit.......

https://github.com/arcadez2003/mame/commit/779c4d6d2d4a818850db8d5d6b88c6754d7c25fe#diff-95279150a39e488eef31ad99de430643a925a0919a1295ff1259ffd4ad1664b4

Notice he removed the speedups and replaced arm7_latch with..
 UINT32 kov2_latchdata_68k_w;
 UINT32 kov2_latchdata_arm_w;

What i did was to keep the speedups with the original arm7_latch when i added the svg prot code
used by demon front and the glad, however i see this call which might be required to disable the prot busy check...

static void kov2_latch_init( void )
{
	//pgm_state *state = machine.driver_data<pgm_state>();

	//state->m_kov2_latchdata_68k_w = 0;
	//state->m_kov2_latchdata_arm_w = 0;

	//state->save_item(NAME(state->m_kov2_latchdata_68k_w));
	//state->save_item(NAME(state->m_kov2_latchdata_arm_w));

	arm7_latch = 0;

}

This is used by all the games in question hopefully by adding a similar call for our arm7_latch
the games will now boot fingers crossed :)
